apps/notifications/src/modules/chain/services/rpc-url-resolver/rpc-url-resolver.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+
+import type { ChainEventsConfig } from "@src/modules/chain/config";
+import { FeatureFlags } from "../feature-flags/feature-flags";
+import { FeatureFlagsService } from "../feature-flags/feature-flags.service";
+
+@Injectable()
+export class RpcUrlResolverService {
+  constructor(
+    private readonly configService: ConfigService<ChainEventsConfig>,
+    private readonly featureFlagsService: FeatureFlagsService
+  ) {}
+
+  async getRpcUrl(): Promise<string> {
```

Comment:
ConfigService has a `set` method. There is also a way to listen for ff changes as done in api already https://github.com/akash-network/console/blob/main/apps/api/src/core/services/feature-flags/feature-flags.service.ts#L48

This means the listener can be done like so

```ts
@Injectable()
export class RpcUrlWatcherService implements OnModuleInit {
  constructor(
    private readonly featureFlagsService: FeatureFlagsService,
    private readonly configService: ConfigService<ChainEventsConfig>
  ) {}

  onModuleInit() {
    this.featureFlagsService.onChanged((...args) => {
      this.configService.set("chain.RPC_NODE_ENDPOINT", some value);
    });
  }
}
```

this way the rest of the app would still use same config service without hustle.

Reply:
I also think when it gets to the config in other apps similar approach can be taken as it's much easier to utilise config service directly from many perspectives

Reply:
I switched it to env var because it was causing too many potential issues live updating the api/rpc urls with many dependencies relying on it. We could clear the instances but that would have unintented consequences.